### PR TITLE
Fix a bug in loading configuration from awsconfiguration.json

### PR DESCRIPTION
**Pull request by kvasukib**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AWSAppSyncClient/AWSAppSyncClient.swift [hunk 1]
```diff
@@ -209,24 +182,48 @@ public class AWSAppSyncClientConfiguration {
...
-        // If AuthType is AWS_IAM, use the AWSCredentialsProvider passed in
-        // or create a provider based on the CognitoIdentity CredentialsProvider
-        // passed from the config
-        var defaultCredentialsProvider: AWSCredentialsProvider? = credentialsProvider
-        if authTypeFromConfig == AuthType.awsIAM {
+            break
+        case AuthType.amazonCognitoUserPools:
+            if credentialsProvider != nil || apiKeyAuthProvider != nil || oidcAuthProvider != nil {
+                throw AWSAppSyncClientInfoError(errorMessage: AuthType.amazonCognitoUserPools.rawValue + " is selected in configuration but a "
+                    + "AWSCognitoUserPoolsAuthProvider object is not passed in.")
```
Comment (rohandubal) [inline AWSAppSyncClient/AWSAppSyncClient.swift]: Should we change the message to indicate that an un-supported auth provider object is passed in?

File: AWSAppSyncClient/AWSAppSyncClient.swift [hunk 2]
```diff
@@ -487,22 +484,26 @@ public class AWSAppSyncClientInfo {
...
+            
+            if let apiKeyFromDictionary = defaultAppSyncConfig["ApiKey"] {
+                self.apiKey = apiKeyFromDictionary as! String
+            } else {
+                if (self.authType == AuthType.apiKey.rawValue) {
+                    throw AWSAppSyncClientInfoError(errorMessage: "API_KEY AuthMode found in configuration but a valid ApiKey is not found")
+                }
+            }
+        } catch {
+            throw AWSAppSyncClientInfoError(errorMessage: "Error in reading AppSync configuration from the awsconfiguration.json/Info.plist")
```
Comment (rohandubal) [inline AWSAppSyncClient/AWSAppSyncClient.swift]: We should remove Info.plist from here (and other places where we have used it)

Comment (kvasukib) [reply AWSAppSyncClient/AWSAppSyncClient.swift]: Thank you for pointing it out. Removed references to Info.plist.

File: AWSAppSyncClient/AWSAppSyncClient.swift (hunk 1 above)
Comment (kvasukib) [reply AWSAppSyncClient/AWSAppSyncClient.swift]: I want to be a little bit more explanatory in the error. Hence explaining it.

**Pull request merged**